mono/metadata/marshal.h

```diff
@@ -21,6 +21,12 @@
 #include <mono/metadata/remoting.h>
 #include <mono/utils/mono-error.h>
 
+#ifndef HOST_WIN32
+typedef gunichar2 *BSTR;
+#else
+// Similar but slightly varying BSTR is provided by windows.h.
```

Comment:
Comment is too alarmist w/o details.
Revise it later.

BSTR in Windows C is unsigned short*, same as here.
In C++ it might be wchar_t, which varies between unsigned short* and a unique type. But BSTR might not be that anyway. But it is close enough. The difference is in C++ overloading and name mangling, which aren't relevant here. Also debugging is better, unsigned short being a small number, native wchar_t being a character.

Reply:
Fixed.